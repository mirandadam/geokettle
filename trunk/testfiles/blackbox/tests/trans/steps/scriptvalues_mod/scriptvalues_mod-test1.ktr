<?xml version="1.0" encoding="UTF-8"?>
<transformation>
  <info>
    <name>script test 1</name>
    <description/>
    <extended_description/>
    <trans_version/>
    <directory>&#47;</directory>
    <log>
      <read/>
      <write/>
      <input/>
      <output/>
      <update/>
      <rejected/>
      <connection/>
      <table/>
      <use_batchid>Y</use_batchid>
      <use_logfield>N</use_logfield>
    </log>
    <maxdate>
      <connection/>
      <table/>
      <field/>
      <offset>0.0</offset>
      <maxdiff>0.0</maxdiff>
    </maxdate>
    <size_rowset>1000</size_rowset>
    <sleep_time_empty>100</sleep_time_empty>
    <sleep_time_full>100</sleep_time_full>
    <unique_connections>N</unique_connections>
    <feedback_shown>Y</feedback_shown>
    <feedback_size>5000</feedback_size>
    <using_thread_priorities>N</using_thread_priorities>
    <shared_objects_file/>
    <dependencies>
    </dependencies>
    <partitionschemas>
    </partitionschemas>
    <slaveservers>
    </slaveservers>
    <clusterschemas>
    </clusterschemas>
  <modified_user>-</modified_user>
  <modified_date>2007&#47;07&#47;24 22:33:59.679</modified_date>
  </info>
  <notepads>
  </notepads>
  <connection>
    <name>Pentaho</name>
    <server>localhost</server>
    <type>MYSQL</type>
    <access>Native</access>
    <database>MBO</database>
    <port>3306</port>
    <username>root</username>
    <password>Encrypted </password>
    <servername/>
    <data_tablespace/>
    <index_tablespace/>
    <attributes>
      <attribute><code>EXTRA_OPTION_MYSQL.defaultFetchSize</code><attribute>500</attribute></attribute>
      <attribute><code>EXTRA_OPTION_MYSQL.useCursorFetch</code><attribute>true</attribute></attribute>
      <attribute><code>IS_CLUSTERED</code><attribute>N</attribute></attribute>
      <attribute><code>MAXIMUM_POOL_SIZE</code><attribute>10</attribute></attribute>
      <attribute><code>PORT_NUMBER</code><attribute>3306</attribute></attribute>
      <attribute><code>STREAM_RESULTS</code><attribute>Y</attribute></attribute>
      <attribute><code>USE_POOLING</code><attribute>N</attribute></attribute>
    </attributes>
  </connection>
  <order>
  <hop> <from>datafile</from><to>script</to><enabled>Y</enabled> </hop>  <hop> <from>script</from><to>XML Output</to><enabled>Y</enabled> </hop>  </order>
  <step>
    <name>XML Output</name>
    <type>XMLOutput</type>
    <description/>
    <distribute>Y</distribute>
    <copies>1</copies>
         <partitioning>
           <method>none</method>
           <schema_name/>
           </partitioning>
    <encoding>UTF-8</encoding>
    <xml_main_element>Rows</xml_main_element>
    <xml_repeat_element>Row</xml_repeat_element>
    <file>
      <name>testfiles&#47;blackbox&#47;tests&#47;trans&#47;steps&#47;scriptvalues_mod&#47;scriptvalues_mod-test1.actual</name>
      <extention>xml</extention>
      <split>N</split>
      <add_date>N</add_date>
      <add_time>N</add_time>
      <zipped>N</zipped>
      <splitevery>0</splitevery>
      </file>
    <fields>
      <field>
        <name>date1</name>
        <element>date1</element>
        <type>Date</type>
        <format>yyyy-MM-dd</format>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Customer_No</name>
        <element>Customer_No</element>
        <type>Integer</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>5</length>
        <precision>0</precision>
        </field>
      <field>
        <name>First_Name</name>
        <element>First_Name</element>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Last_Name</name>
        <element>Last_Name</element>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Address</name>
        <element>Address</element>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Birth_Date</name>
        <element>Birth_Date</element>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Full_Name</name>
        <element>Full_Name</element>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>30</length>
        <precision>0</precision>
        </field>
      <field>
        <name>Customer_No2</name>
        <element/>
        <type>Integer</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>date2</name>
        <element/>
        <type>Date</type>
        <format>yyyy-MM-dd</format>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>year</name>
        <element/>
        <type>Integer</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <nullif/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      </fields>
     <cluster_schema/>
 <remotesteps>   <input>   </input>   <output>   </output> </remotesteps>    <GUI>
      <xloc>399</xloc>
      <yloc>109</yloc>
      <draw>Y</draw>
      </GUI>
    </step>

  <step>
    <name>datafile</name>
    <type>CsvInput</type>
    <description/>
    <distribute>Y</distribute>
    <copies>1</copies>
         <partitioning>
           <method>none</method>
           <schema_name/>
           </partitioning>
    <filename>testfiles&#47;customer-dimension-test.csv</filename>
    <separator>;</separator>
    <enclosure>&quot;</enclosure>
    <header>Y</header>
    <buffer_size>30</buffer_size>
    <lazy_conversion>N</lazy_conversion>
    <fields>
      <field>
        <name>date1</name>
        <type>Date</type>
        <format>dd-MM-yyyy</format>
        <currency/>
        <decimal/>
        <group/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Customer_No</name>
        <type>Integer</type>
        <format>#</format>
        <currency/>
        <decimal/>
        <group/>
        <length>5</length>
        <precision>0</precision>
        </field>
      <field>
        <name>First_Name</name>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Last_Name</name>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Address</name>
        <type>String</type>
        <format/>
        <currency/>
        <decimal/>
        <group/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      <field>
        <name>Birth_Date</name>
        <type>String</type>
        <format>dd-MM-yyyy</format>
        <currency/>
        <decimal/>
        <group/>
        <length>-1</length>
        <precision>-1</precision>
        </field>
      </fields>
     <cluster_schema/>
 <remotesteps>   <input>   </input>   <output>   </output> </remotesteps>    <GUI>
      <xloc>129</xloc>
      <yloc>109</yloc>
      <draw>Y</draw>
      </GUI>
    </step>

  <step>
    <name>script</name>
    <type>ScriptValueMod</type>
    <description/>
    <distribute>Y</distribute>
    <copies>1</copies>
         <partitioning>
           <method>none</method>
           <schema_name/>
           </partitioning>
    <script/>
    <compatible>Y</compatible>
    <jsScripts>      <jsScript>        <jsScript_type>0</jsScript_type>
        <jsScript_name>Script 1</jsScript_name>
        <jsScript_script>
var Full_Name = First_Name.getString() + &quot; &quot; +Last_Name.getString();
var CustomerNo2 = Customer_No.getInteger() + 1;
var aDate = date1.getDate();
var aYear = aDate.getYear()+1900;
</jsScript_script>
      </jsScript>    </jsScripts>    <fields>      <field>        <name>Full_Name</name>
        <rename>Full_Name</rename>
        <type>String</type>
        <length>30</length>
        <precision>0</precision>
      </field>      <field>        <name>CustomerNo2</name>
        <rename>Customer_No2</rename>
        <type>Integer</type>
        <length>-1</length>
        <precision>-1</precision>
      </field>      <field>        <name>aDate</name>
        <rename>date2</rename>
        <type>Date</type>
        <length>-1</length>
        <precision>-1</precision>
      </field>      <field>        <name>aYear</name>
        <rename>year</rename>
        <type>Integer</type>
        <length>-1</length>
        <precision>-1</precision>
      </field>    </fields>     <cluster_schema/>
 <remotesteps>   <input>   </input>   <output>   </output> </remotesteps>    <GUI>
      <xloc>264</xloc>
      <yloc>109</yloc>
      <draw>Y</draw>
      </GUI>
    </step>

  <step_error_handling>
  </step_error_handling>
</transformation>
