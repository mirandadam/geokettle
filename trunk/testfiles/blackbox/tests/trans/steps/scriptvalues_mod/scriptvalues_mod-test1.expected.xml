<?xml version="1.0" encoding="UTF-8"?>
<Rows>
 <Row><date1>1996-10-01</date1> <Customer_No> 10000</Customer_No> <First_Name>Matt</First_Name> <Last_Name>\&quot;Casters\&quot;</Last_Name> <Address>Pekelharing 32, 9000 GENT</Address> <Birth_Date>1-1-1970</Birth_Date> <Full_Name>Matt \&quot;Casters\&quot;</Full_Name> <Customer_No2> 10001</Customer_No2> <date2>1996-10-01</date2> <year> 1996</year> </Row>
 <Row><date1>2000-08-01</date1> <Customer_No> 10000</Customer_No> <First_Name>Matt</First_Name> <Last_Name>Casters</Last_Name> <Address>Assesteenweg 234, 1742 TERNAT</Address> <Birth_Date/> <Full_Name>Matt Casters</Full_Name> <Customer_No2> 10001</Customer_No2> <date2>2000-08-01</date2> <year> 2000</year> </Row>
 <Row><date1>2002-05-01</date1> <Customer_No> 10000</Customer_No> <First_Name>Matt</First_Name> <Last_Name>Casters</Last_Name> <Address>Fonteinstraat 70, 9400 OKEGEM</Address> <Birth_Date>14-2-1969</Birth_Date> <Full_Name>Matt Casters</Full_Name> <Customer_No2> 10001</Customer_No2> <date2>2002-05-01</date2> <year> 2002</year> </Row>
</Rows>
